Add threads tests

jyapayne opened the pull request:
I mainly just want some feedback on the test style before I write more. Let me know if some things could change and if modifying the frontend for the tests is okay.

dom96 commented on tests/browsertests/threads.nim:
```diff
@@ -0,0 +1,46 @@
+    let newThreadTitle = session.findElement("#thread-title")
+    check newThreadTitle.isSome()
+
+    check newThreadTitle.get().getText() == "This is a thread title!"
```
Not a big deal but I would get rid of this duplication by placing the title in a ``const``.

dom96 replied:
Can do that in your next PR ;D

jyapayne replied:
Will do!

jyapayne commented:
> I'm also wondering, and this is an open question, whether it would make sense to prefix the HTML IDs we are introducing with "test" or something like that to make it clear they are used for testing and not the functionality of the website. What do you think?

Personally, I think it's okay to leave the ones I added as they are. Prefixing with test doesn't gain much unless it's very clearly only required for testing. The ids and classes I added are pretty standard fair for many websites and can be used in the future for styling. For example, since I added the "original-post" class, you could add some css to that and make the original post stand out somehow. Of course, if someone doesn't know and removes that class later, the test will tell them. So I think it's fine :)

jyapayne commented:
Thanks for the merge and the review!
